[BUG] Linker Error when using Qt6

## Description

Trying to build with Qt6 using cmake results in a linker error.

## Prerequisites

If applicable to the issue, here are a few things you should provide to help me understand the issue:

- Qt version: 6
- Extension: None

## Test case

```
$ git clone git@github.com:Alexhuszagh/BreezeStyleSheets.git
$ cd BreezeStyleSheets/example/cmake/
$ vim CMakeList.txt
$ git diff

diff --git a/example/cmake/CMakeLists.txt b/example/cmake/CMakeLists.txt
index 83a0bd3..df53314 100644
--- a/example/cmake/CMakeLists.txt
+++ b/example/cmake/CMakeLists.txt
@@ -4,7 +4,7 @@ cmake_policy(SET CMP0128 NEW)
 project(testing)
 set(CMAKE_CXX_STANDARD 17)
 
-set(QT_VERSION Qt5 CACHE STRING "The Qt version framework to use (Qt5 or Qt6).")
+set(QT_VERSION Qt6 CACHE STRING "The Qt version framework to use (Qt5 or Qt6).")
 set(BREEZE_EXTENSIONS all CACHE STRING "The extensions to include in our stylesheets.")
 set(BREEZE_STYLES all CACHE STRING "The styles to include in our stylesheets.")

$ cmake -B build -S . # (success)
$ cmake --build build
[  8%] Generating themes
[  8%] Built target run_python_breeze
[ 16%] Automatic RCC for _deps/breeze_stylesheets-src/dist/breeze.qrc
[ 16%] Built target breeze_arcc_breeze
[ 16%] Built target breeze_autogen_timestamp_deps
[ 25%] Automatic MOC and UIC for target breeze
[ 25%] Built target breeze_autogen
[ 33%] Building CXX object CMakeFiles/breeze.dir/breeze_autogen/mocs_compilation.cpp.o
[ 41%] Building CXX object CMakeFiles/breeze.dir/breeze_autogen/7VP5XYJABG/qrc_breeze.cpp.o
[ 50%] Linking CXX static library libbreeze.a
[ 58%] Built target breeze
[ 58%] Built target testing_autogen_timestamp_deps
[ 66%] Automatic MOC and UIC for target testing
[ 66%] Built target testing_autogen
[ 75%] Building CXX object CMakeFiles/testing.dir/testing_autogen/mocs_compilation.cpp.o
[ 83%] Building CXX object CMakeFiles/testing.dir/test_qt.cpp.o
[ 91%] Linking CXX executable testing
/usr/bin/ld: libbreeze.a(qrc_breeze.cpp.o): copy relocation against non-copyable protected symbol `qt_resourceFeatureZstd@@Qt_6' in /usr/lib/libQt6Core.so.6.8.2
collect2: Fehler: ld gab 1 als Ende-Status zurück
make[2]: *** [CMakeFiles/testing.dir/build.make:130: testing] Fehler 1
make[1]: *** [CMakeFiles/Makefile2:169: CMakeFiles/testing.dir/all] Fehler 2
make: *** [Makefile:91: all] Fehler 2
```

## Environment:

```
$ uname -a
Linux EsGehArch 6.13.1-arch1-1 #1 SMP PREEMPT_DYNAMIC Sun, 02 Feb 2025 01:02:29 +0000 x86_64 GNU/Linux

$ cmake --version
cmake version 3.31.6

CMake suite maintained and supported by Kitware (kitware.com/cmake).
```

--- a/README.md
+++ b/README.md
@@ -220,6 +220,12 @@ int main(int argc, char *argv[])
 }
 ```
 
+If you get an issue with copy relocations against a non-copyable protected [symbol](https://github.com/Alexhuszagh/BreezeStyleSheets/issues/121), ensure the `POSITION_INDEPENDENT_CODE` is set to `ON`. For the example above, add the following line:
+
+```cmake
+set_property(TARGET myapp PROPERTY POSITION_INDEPENDENT_CODE ON)
+```
+
 ### QMake Installation
 
 Copy the contents of the `dist` subdirectory into your project directory and add the qrc file to your project file.

--- a/example/cmake/CMakeLists.txt
+++ b/example/cmake/CMakeLists.txt
@@ -1,10 +1,12 @@
-cmake_minimum_required(VERSION 3.13)
+cmake_minimum_required(VERSION 3.14)
 cmake_policy(SET CMP0128 NEW)
+cmake_policy(SET CMP0083 NEW)
 
 project(testing)
 set(CMAKE_CXX_STANDARD 17)
 
 set(QT_VERSION Qt5 CACHE STRING "The Qt version framework to use (Qt5 or Qt6).")
+set(PREFER_PIE OFF CACHE BOOL "If to prefer position-independent code, if the compiler supports it.")
 set(BREEZE_EXTENSIONS all CACHE STRING "The extensions to include in our stylesheets.")
 set(BREEZE_STYLES all CACHE STRING "The styles to include in our stylesheets.")
 
@@ -12,3 +14,12 @@ include(${CMAKE_CURRENT_SOURCE_DIR}/breeze.cmake)
 set(SOURCE_FILES ${CMAKE_CURRENT_SOURCE_DIR}/test_qt.cpp)
 add_executable(testing ${SOURCE_FILES})
 target_link_libraries(testing PRIVATE Qt${QT_VERSION_MAJOR}::Widgets breeze)
+
+# Prefer position independent code, if the compiler supports it.
+if(PREFER_PIE)
+  include(CheckPIESupported)
+  check_pie_supported(LANGUAGES CXX OUTPUT_VARIABLE pie-supported)
+  if(CMAKE_C_LINK_PIE_SUPPORTED)
+    set_property(TARGET breeze PROPERTY POSITION_INDEPENDENT_CODE ON)
+  endif()
+endif()

